src/basic/time-util.c

```diff
@@ -847,19 +847,23 @@ int parse_timestamp(const char *t, usec_t *usec) {
 
 from_tm:
         x = mktime_or_timegm(&tm, utc);
-        if (x < 0)
-                return -EINVAL;
+        if (x == (time_t) -1)
+                return -EOVERFLOW;
```

Comment:
uh, i am not sure i like this. this creates an ambiguity: a negative time_t is permitted, except if it happens to be -1, in which case it is refused. (or in other words: "1969-12-31 23:59:59" will fail to parse but "1969-12-31 23:59:58" and "1970-01-01 00:00:00" will succeed)

I think we should really fail in all cases of "negative" times here. For our purposes times before 1970 make very little sense, and we should just cleanly error out on all of them.